host/http.go

```diff
@@ -228,6 +230,45 @@ func (h *jobAPI) GetStatus(w http.ResponseWriter, r *http.Request, _ httprouter.
 	httphelper.JSON(w, 200, &h.host.status)
 }
 
+func checkPort(port host.Port) bool {
+	fmt.Println("Checking port", port.Port)
+	l, err := net.Listen(port.Proto, fmt.Sprintf(":%d", port.Port))
+	if err != nil {
+		return false
+	}
+	l.Close()
+	return true
+}
+
+func (h *jobAPI) ResourceCheck(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	var ports []host.Port
+	conflicts := make([]host.Port, 0)
+	if err := httphelper.DecodeJSON(r, &ports); err != nil {
```

Comment:
If this is a "resource check" and not a "port check" it should use a top-level struct for extensibility.